doc/changelog/14-misc/20809-autonaming-goals-Added.rst

```diff
@@ -0,0 +1,4 @@
+- **Added:**
+  Goal names can be automatically generated for `induction`, `destruct` and `eapply`
```

Comment:
Please add a hyperlink to the new option before mentioning the tactics.  Also, enclose the tactic names in ``` :tacn:`...` ```